datarepo-clienttests/src/main/java/utils/KubernetesClientUtils.java

```diff
@@ -236,6 +240,44 @@ public static V1Deployment changeReplicaSetSize(V1Deployment deployment, int num
             null);
   }
 
+  /** Select any pod from api pods and delete pod. */
+  public static void deleteRandomPod() throws ApiException, IOException {
+    V1Deployment apiDeployment = KubernetesClientUtils.getApiDeployment();
+    if (apiDeployment == null) {
+      throw new RuntimeException("API deployment not found.");
+    }
+    printApiPodCount(apiDeployment, "Before deleting pods");
+    printApiPods(apiDeployment);
+    String deploymentComponentLabel = apiDeployment.getMetadata().getLabels().get(componentLabel);
+
+    // select a "random" pod from list of apis
+    String randomPodName;
+    randomPodName =
+        listPods().stream()
+            .filter(
+                pod ->
+                    deploymentComponentLabel.equals(
+                        pod.getMetadata().getLabels().get(componentLabel)))
+            .findAny()
```

Comment:
Is that really random? The documentation says non deterministic, but i'm not sure that's the same thing.
Could you tell in your testing? The alternative would be to pick a random index in the pod list.

Reply:
Ok - I made a change to make it closer to random. Let me know what you think!